diff --git a/av/include/gz/common/Video.hh b/av/include/gz/common/Video.hh
index 775d09d..15fbe92 100644
--- a/av/include/gz/common/Video.hh
+++ b/av/include/gz/common/Video.hh
@@ -17,8 +17,9 @@
 #ifndef GZ_COMMON_VIDEO_HH_
 #define GZ_COMMON_VIDEO_HH_
 
-#include <string>
+#include <chrono>
 #include <memory>
+#include <string>
 
 #include <gz/common/config.hh>
 #include <gz/common/av/Export.hh>
diff --git a/events/include/gz/common/Event.hh b/events/include/gz/common/Event.hh
index 471c582..557993e 100644
--- a/events/include/gz/common/Event.hh
+++ b/events/include/gz/common/Event.hh
@@ -18,6 +18,7 @@
 #define GZ_COMMON_EVENT_HH_
 
 #include <atomic>
+#include <chrono>
 #include <functional>
 #include <list>
 #include <map>
diff --git a/graphics/include/gz/common/Animation.hh b/graphics/include/gz/common/Animation.hh
index a70e05a..0fea0dc 100644
--- a/graphics/include/gz/common/Animation.hh
+++ b/graphics/include/gz/common/Animation.hh
@@ -17,6 +17,7 @@
 #ifndef IGNITION_COMMON_ANIMATION_HH_
 #define IGNITION_COMMON_ANIMATION_HH_
 
+#include <chrono>
 #include <map>
 #include <string>
 #include <vector>
diff --git a/include/gz/common/Timer.hh b/include/gz/common/Timer.hh
index a2a768c..aee527c 100644
--- a/include/gz/common/Timer.hh
+++ b/include/gz/common/Timer.hh
@@ -17,6 +17,8 @@
 #ifndef IGNITION_COMMON_TIMER_HH_
 #define IGNITION_COMMON_TIMER_HH_
 
+#include <chrono>
+
 #include <gz/common/config.hh>
 #include <gz/common/Time.hh>
 #include <gz/common/Export.hh>
diff --git a/include/gz/common/WorkerPool.hh b/include/gz/common/WorkerPool.hh
index 81ec85f..d0ee6e8 100644
--- a/include/gz/common/WorkerPool.hh
+++ b/include/gz/common/WorkerPool.hh
@@ -18,6 +18,7 @@
 #ifndef IGNITION_COMMON_WORKER_POOL_HH_
 #define IGNITION_COMMON_WORKER_POOL_HH_
 
+#include <chrono>
 #include <functional>
 #include <memory>
 
diff --git a/test/integration/encoder_timing.cc b/test/integration/encoder_timing.cc
index 365ff84..530695a 100644
--- a/test/integration/encoder_timing.cc
+++ b/test/integration/encoder_timing.cc
@@ -13,8 +13,10 @@
  * limitations under the License.
  *
 */
-#include <gtest/gtest.h>
 #include <array>
+#include <chrono>
+#include <gtest/gtest.h>
+
 #include "gz/common/VideoEncoder.hh"
 #include "gz/common/Video.hh"
 #include "test_config.h"
diff --git a/test/integration/video_encoder.cc b/test/integration/video_encoder.cc
index 278b6ed..226346f 100644
--- a/test/integration/video_encoder.cc
+++ b/test/integration/video_encoder.cc
@@ -15,7 +15,8 @@
 */
 #include <gtest/gtest.h>
 
-// needed on MacOS
+// needed on MacOS and Windows
+#include <chrono>
 #include <cmath>
 
 #include "gz/common/VideoEncoder.hh"
diff --git a/test/performance/logging.cc b/test/performance/logging.cc
index 19d5499..dd8976c 100644
--- a/test/performance/logging.cc
+++ b/test/performance/logging.cc
@@ -17,6 +17,7 @@
 #include <gtest/gtest.h>
 
 #include <atomic>
+#include <chrono>
 #include <map>
 #include <thread>
 
